business/controlplane_monitor.go

```diff
@@ -110,12 +115,22 @@ func (p *controlPlaneMonitor) RefreshIstioCache(ctx context.Context) error {
 		if len(controlPlanes) > 0 {
 			// Since it doesn't matter what revision we choose, just choose the first one.
 			controlPlane := controlPlanes[0]
+			if controlPlane.Status != kubernetes.ComponentHealthy {
+				log.Warningf("Skipping controlplane [%s] in cluster [%s] because it is not healthy.", controlPlane.Revision, cluster)
+				if controlPlane.Status == kubernetes.ComponentUnreachable {
+					log.Warningf("unable to proxy Istiod pods. " +
+						"Make sure your Kubernetes API server has access to the Istio control plane through 8080 port")
```

Comment:
These warning messages are identical to the ones about 20 lines above. So when you look at the log files and happen to see either `Skipping controlplane [xx] in cluster [yy] because it is not healthy.` or `unable to proxy Istiod pods. Make sure your Kubernetes API server has access to the Istio control plane through 8080 port` you will never know which of these two places in the code generated those log messages, which could make debugging more difficult than it has to be.

My suggestion is to never have multiple places in code with identical log message wording. Alter it even slightly, just to make it easier to differentiate which place in the code generated the log message. e.g.
```suggestion
				log.Warningf("After choosing first revision - Skipping controlplane [%s] in cluster [%s] because it is not healthy.", controlPlane.Revision, cluster)
				if controlPlane.Status == kubernetes.ComponentUnreachable {
					log.Warningf("After choosing first revision - unable to proxy Istiod pods. " +
						"Make sure your Kubernetes API server has access to the Istio control plane through 8080 port")
```